enterprise/trackedusers/data_collector.go

```diff
@@ -0,0 +1,192 @@
+package trackedusers
+
+//go:generate mockgen -destination=./mocks/mock_data_collector.go -package=mockdatacollector github.com/rudderlabs/rudder-server/enterprise/trackedusers DataCollector
```

Comment:
I would avoid using mockgen for now. 

It has simple enough interface that makes it easy to create a custom mock. Usually, the generated mock is not very convenient to use, and leading to partial testing in practise.

A good mock for the MTU collector could count the number of unique IDs per workspace per anon/user/identify call. Thus instead of asserting if method was called with the correct arguments, we can test the outcome. 

Also consider exposing a NOOP implementation, it would be useful for disabling the functionality and for tests that don't require testing mtu.

Reply:
in this particular using mocks makes more sense as we just want to make sure that this methods are correctly called with correct parameters in unit tests, additional logic can be tested in integration tests.